app/controllers/incomeSources/cease/CeaseForeignPropertyController.scala

```diff
@@ -43,42 +44,40 @@ class CeaseForeignPropertyController @Inject()(val authenticate: AuthenticationP
+                                               val customNotFoundErrorView: CustomNotFoundError,
+                                               val sessionService: SessionService)
                                               (implicit val appConfig: FrontendAppConfig,
                                                mcc: MessagesControllerComponents,
                                                val ec: ExecutionContext,
                                                val itvcErrorHandler: ItvcErrorHandler,
                                                val itvcErrorHandlerAgent: AgentItvcErrorHandler
                                               )
-  extends ClientConfirmedController with FeatureSwitching with I18nSupport {
+  extends ClientConfirmedController with FeatureSwitching with I18nSupport with IncomeSourcesUtils {
 
   def handleRequest(isAgent: Boolean, origin: Option[String] = None)
-                   (implicit user: MtdItUser[_], hc: HeaderCarrier, ec: ExecutionContext, messages: Messages): Future[Result] = {
+                   (implicit user: MtdItUser[_], hc: HeaderCarrier, ec: ExecutionContext, messages: Messages): Future[Result] = withIncomeSourcesFS {
 
-    val incomeSourcesEnabled: Boolean = isEnabled(IncomeSources)
     val backUrl: String = if (isAgent) controllers.incomeSources.cease.routes.CeaseIncomeSourceController.showAgent().url else
       controllers.incomeSources.cease.routes.CeaseIncomeSourceController.show().url
     val postAction: Call = if (isAgent) controllers.incomeSources.cease.routes.CeaseForeignPropertyController.submitAgent else
       controllers.incomeSources.cease.routes.CeaseForeignPropertyController.submit
-    val errorHandler: ShowInternalServerError = if (isAgent) itvcErrorHandlerAgent else itvcErrorHandler
-
-    if (incomeSourcesEnabled) {
-      Future.successful(Ok(view(
-        ceaseForeignPropertyForm = CeaseForeignPropertyForm.form,
-        postAction = postAction,
-        isAgent = isAgent,
-        backUrl = backUrl,
-        origin = origin)(user, messages)))
-    } else {
-      Future.successful(Ok(customNotFoundErrorView()(user, messages)))
-    } recover {
-      case ex: Exception =>
-        Logger("application").error(s"${if (isAgent) "[Agent]"}" +
-          s"Error getting CeaseForeignProperty page: ${ex.getMessage}")
-        errorHandler.showInternalServerError()
-    }
+
+    Future.successful(Ok(view(
+      ceaseForeignPropertyForm = CeaseForeignPropertyForm.form,
+      postAction = postAction,
+      isAgent = isAgent,
+      backUrl = backUrl,
+      origin = origin)(user, messages)))
+
+  } recover {
+    case ex: Exception =>
+      val errorHandler: ShowInternalServerError = if (isAgent) itvcErrorHandlerAgent else itvcErrorHandler
+      Logger("application").error(s"${if (isAgent) "[Agent]"}" +
+        s"Error getting CeaseForeignProperty page: ${ex.getMessage}")
```

Comment:
Traceability could be improved by adding:
[CeaseForeignPropertyController][handleRequest]